auth/certificates/master.go

```diff
@@ -0,0 +1,401 @@
+package certificates
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/bsv-blockchain/go-sdk/overlay"
+	ec "github.com/bsv-blockchain/go-sdk/primitives/ec"
+	"github.com/bsv-blockchain/go-sdk/wallet"
+)
+
+var (
+	ErrInvalidMasterCertificate = errors.New("invalid master certificate")
+	ErrMissingMasterKeyring     = errors.New("master keyring is required")
+	ErrFieldNotFound            = errors.New("field not found")
+	ErrKeyNotFoundInKeyring     = errors.New("key not found in keyring")
+	ErrDecryptionFailed         = errors.New("decryption failed")
+	ErrEncryptionFailed         = errors.New("encryption failed")
+)
+
+// MasterCertificate extends the Certificate struct to include a master keyring
+// for key management and selective disclosure of certificate fields.
+// It mirrors the structure and functionality of the MasterCertificate class in the TypeScript SDK.
+type MasterCertificate struct {
+	// Embed the base Certificate struct
+	Certificate
+	// MasterKeyring contains encrypted symmetric keys (Base64 encoded) for each field.
+	// The key is the field name, and the value is the encrypted key.
+	MasterKeyring map[wallet.CertificateFieldNameUnder50Bytes]wallet.Base64String `json:"masterKeyring,omitempty"`
+}
+
+// NewMasterCertificate creates a new MasterCertificate instance.
+// It validates that the masterKeyring contains an entry for every field in the base certificate.
+func NewMasterCertificate(
+	cert *Certificate,
+	masterKeyring map[wallet.CertificateFieldNameUnder50Bytes]wallet.Base64String,
+) (*MasterCertificate, error) {
+	if masterKeyring == nil {
```

Comment:
As we operate on map here I think better solution would be:
```suggestion
	if len(masterKeyring) == 0 {
```